djangocms_versioning/admin.py

```diff
@@ -412,6 +444,29 @@ def edit_redirect_view(self, request, object_id):
             ), args=(version.content.pk,))
         return redirect(url)
 
+    def revert_redirect_view(self, request, object_id):
+        """Redirects to the admin change view and creates a draft version
+        if no draft exists yet.
+        """
+        # This view always changes data so only POST requests should work
+        version = self.get_object(request, unquote(object_id))
+        version = version.copy(request.user)
```

Comment:
This happens too fast (before `version is None` check). Also you need to check if you're reverting from an archived/unpublished version, and that there's no existing drafts